### PR TITLE
Add Logical Color Combinations

### PR DESCRIPTION
**Pull request by llPlayll**

### CODE REVIEW

File: Assets/ModScripts/Submodules/LogicalColorCombinations.cs [hunk 1]
```diff
@@ -0,0 +1,316 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Security.Principal;
+using UnityEngine;
+using KModkit;
+using Random = UnityEngine.Random;
```
Comment (Poncedeleon100) [inline Assets/ModScripts/Submodules/LogicalColorCombinations.cs:9]:
```suggestion
﻿using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using UnityEngine;
using Random = UnityEngine.Random;
```
Some of these usings aren't necessary

File: Assets/ModScripts/Submodules/LogicalColorCombinations.cs [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+    bool submissionMode = false;
+    bool struckThisStage;
+    int curLEDPos = -1;
+    int curLEDColor = Random.Range(0, 9);
+    char inputChar;
+    int targetColor, targetDir = 0;
+    List<int> solvedLEDs = new List<int>();
+    List<int> solvedColors = new List<int>();
+
+    public LogicalColorCombinations(CruelModkitScript Module, int ModuleID, ComponentInfo Info, bool Vanilla, byte Components) : base(Module, ModuleID, Info, Vanilla, Components)
```
Comment (Poncedeleon100) [inline Assets/ModScripts/Submodules/LogicalColorCombinations.cs:74]:
```suggestion
    public LogicalColorCombinations(CruelModkitScript Module, int ModuleID, ComponentInfo Info, byte Components) : base(Module, ModuleID, Info, Components)
```
Remove the Vanilla bool

File: Assets/ModScripts/Submodules/LogicalColorCombinations.cs [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+    readonly string[] operatorNames = { "AND", "OR", "XOR", "NAND", "NOR", "XNOR" };
+    readonly string base36 = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    readonly string[] ArrowDirections = { "Up", "Down", "Left", "Right" };
+    int[] flashes = new int[8];
+    string[] LogABCD = new string[4];
+    int[] ABCD = new int[4];
+    int logicOperator;
+    string pairOrder = "";
+    string[] operationPairs = new string[2];
+    string[] finalPairs = new string[2];
```
Comment (Poncedeleon100) [inline Assets/ModScripts/Submodules/LogicalColorCombinations.cs:63]: These fields can all be readonly, along with the solvedLEDs and solvedColors lists

File: Manual/HTML/Logical Color Combinations.html [hunk 4]
```diff
@@ -0,0 +1,450 @@
+﻿<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="initial-scale=1">
+    <title>Logical Color Combinations — Keep Talking and Nobody Explodes Module</title>
+    <link rel="stylesheet" type="text/css" href="css/font.css">
+    <link rel="stylesheet" type="text/css" href="Css/normalize.css">
```
Comment (Poncedeleon100) [inline Manual/HTML/Logical Color Combinations.html:8]:
```suggestion
    <link rel="stylesheet" type="text/css" href="css/normalize.css">
```
This shouldn't be capitalized

File: Manual/HTML/Logical Color Combinations.html [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+    <script src="js/Utilities/svg-utils.js"></script>
+    <style>
+        .dark table, table {
+            margin: 1em auto;
+            border: none;
+        }
+
+        .led-table { text-align: center; }
+        .operator-table { text-align: center; }
+        .operator-table td { font-size: 80%; }
```
Comment (Poncedeleon100) [inline Manual/HTML/Logical Color Combinations.html:21]: These custom classes should follow the following naming convention:
`.module-name-class-name`

File: Manual/HTML/Logical Color Combinations.html [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+                    </tr>
+                    <tr>
+                        <th>RED</th>
+                        <th>YELLOW</th>
+                        <th>BLUE</th>
+                        <th>GREEN</th>
+                        <th>RED</th>
+                        <th>YELLOW</th>
+                        <th>BLUE</th>
+                        <th>GREEN</th>
```
Comment (Poncedeleon100) [inline Manual/HTML/Logical Color Combinations.html:57]: These cells could use the actual color as the background color so it looks a bit nicer (like in Metered Button and Who's Who's manuals). If you do add this, make sure that you use the same classes used in the other manuals but add another one for grey

File: Manual/HTML/Logical Color Combinations.html [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+                        <td>↓<br>CD/AB</td>
+                        <td>|<br>AB/CD</td>
+                        <td>↓<br>AD/BC</td>
+                        <td>∧<br>AC/BD</td>
+                    </tr>
+                </table>
+                <ui>
+                    <li>Apply the logic gate to each letter pair to get 2 binary numbers.</li>
+                    <li>Convert both binary numbers to Base 36. Prepend 0 until they are 3 digits long. Concatenate the numbers.</li>
+                </ui>
```
Comment (Poncedeleon100) [inline Manual/HTML/Logical Color Combinations.html:345]: Replace the `<ui>` tag with the `<ul>` tag throughout the document

**Review by Poncedeleon100 — CHANGES_REQUESTED** [review]

**Comment by Gameplushy** [comment]
In the manual, use the empty set symbol instead of 0*

**Review by Poncedeleon100 — CHANGES_REQUESTED** [review]

**Review by Poncedeleon100 — APPROVED** [review]

**Pull request merged**